core-bundle/src/Controller/Page/RedirectPageController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller\Page;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsPage;
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+use Contao\CoreBundle\Util\UrlUtil;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+#[AsPage('redirect', '')]
+class RedirectPageController
+{
+    public function __construct(private readonly InsertTagParser $insertTagParser)
+    {
+    }
+
+    public function __invoke(Request $request, PageModel $pageModel): Response
+    {
+        $status = 'temporary' === $pageModel->redirect ? Response::HTTP_SEE_OTHER : Response::HTTP_MOVED_PERMANENTLY;
+        $url = $this->insertTagParser->replaceInline($pageModel->url);
+        $url = UrlUtil::makeAbsolute($url, $request->getBasePath());
```

Comment:
```suggestion
        $url = UrlUtil::makeAbsolute($url, $request->getBasePath().'/');
```

Reply:
In the current `PageRedirect` class we only use `Environment::get('base')` though 🤔

Reply:
Environment::get('base') adds the slash itself (and also includes scheme and host)

Reply:
Hm, is `Environment::get('base')` equivalent to `$request->getUriForPath('/')` then?

Reply:
No, `$request->getUriForPath('/')` would get `https://example.com/subdir/index.php/` instead of `https://example.com/subdir/` if you dont use URL rewriting.

`Environment::get('base')` is equivalent to `$request->getSchemeAndHttpHost().$request->getBasePath().'/'`

Reply:
Hm, but if you access

```
https://example.com/foobar/index.php/foobar
```

where `foobar` is the external redirect and the URL entered in the redirect page is `lorem/ipsum`, where should it redirect to?

```
https://example.com/foobar/index.php/lorem/ipsum
```

or

```
https://example.com/foobar/lorem/ipsum
```

? With `$request->getUriForPath('/')` the former would be the case, which I think is correct.

Reply:
If the URL is retrieved from an insert tag it would already be `index.php/lorem/ipsum` so `$request->getUriForPath('/')` would redirect you to https://example.com/foobar/index.php/index.php/lorem/ipsum which is wrong IMO.

The convention in Contao is that every relative URL is relative to `Environment::get('base')` IMO.

Reply:
Maybe we should add a serive that handles all that with methods like:

```php
toAbsoluteUrl(string $url): string;
toPathAbsoluteUrl(string $url): string;
```

This service could then fall back to the request context if no request is available in the request stack.

Reply:
> The insert tag would return `/foobar/index.php/lorem/ipsum`

Yes, as this is for Contao 5 (sorry, I missed that initially), but it still holds for any third party insert tag IMO.

Don’t you think _“The convention in Contao is that every relative URL is relative to `Environment::get('base')`”_ is correct?

Reply:
I do think so :+1: Just wanted to explore (again) whether we can use one of the existing methods.

Regarding the service: I am not sure. Woludn't that be an `UrlUtilUtil` service? 😁

Reply:
> Regarding the service: I am not sure. Woludn't that be an `UrlUtilUtil` service? 😁

I’d think something like `Contao\CoreBundle\Routing\AbsoluteUrlConverter` alias `@contao.routing.absolute_url_converter`

I’d leave the `UrlUtil` class independent of it as it is not related to Contao and just follows the URL standard.

Reply:
Now using `UrilUtil::makeAbsolutePath` from #6569